write-fonts/src/tables/variations.rs

```diff
@@ -129,25 +129,30 @@ impl PackedDeltas {
 
         /// compute the number of deltas in the next run, and whether they are i8s or not
         fn next_run_len(slice: &[i16]) -> (usize, bool) {
-            let (first, rest) = slice.split_first().expect("bounds checked before here");
-            debug_assert!(*first != 0);
-            let is_1_byte = in_i8_range(*first);
+            let first = *slice.first().expect("bounds checked before here");
+            debug_assert!(first != 0);
+            let is_1_byte = in_i8_range(first);
 
-            for (i, raw) in rest.iter().copied().enumerate().take(MAX_POINTS_PER_RUN) {
-                let two_zeros = raw == 0 && rest.get(i + 1) == Some(&0);
-                let different_enc_len = in_i8_range(raw) != is_1_byte;
+            let mut idx = 1;
+            while idx < MAX_POINTS_PER_RUN && idx < slice.len() {
+                let cur = slice[idx];
+
+                // Any reason to stop?
+                let two_zeros = cur == 0 && slice.get(idx + 1) == Some(&0);
+                let different_enc_len = in_i8_range(cur) != is_1_byte;
                 if two_zeros || different_enc_len {
-                    return (i + 1, is_1_byte);
+                    break;
                 }
-            }
 
-            (slice.len(), is_1_byte)
+                idx += 1;
+            }
+            (idx, is_1_byte)
         }
 
         let mut deltas = self.deltas.as_slice();
 
         std::iter::from_fn(move || {
-            if *deltas.first()? == 0 {
+            let run = if *deltas.first()? == 0 {
```

Comment:
Just curious: did clippy complain here or do you just find it more clear to create a local?